### PR TITLE
Simplify the application domain model and improve overall design

### PR DESCRIPTION
**Pull request by peterschutt**
- overhaul of the repository pattern and service objects
- separation of the controllers from the domain objects
- introduction of the SAQ worker
- uses the sqlalchemy plugin

### CODE REVIEW

**Comment by peterschutt** [comment]
some todos:

- [ ] repository not wrapping all sqlalchemy exceptions anymore
- [ ] work out why the controllers aren't documenting
- [ ] remove the authorization docs from routes that are excluded
- [ ] generate the initial migrations
- [ ] ...

File: app/controllers/authors.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+
+def provides_service(db_session: AsyncSession, user: User) -> Service:
+    """Constructs repository and service objects for the request."""
+    return Service(Repository(session=db_session), user)
+
+
+@get()
+async def get_authors(service: Service) -> list[Author]:
+    """Get list of authors."""
```
Comment (Goldziher) [inline app/controllers/authors.py:22]: a list

File: app/controllers/authors.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    return await service.get(author_id)
+
+
+@put(DETAIL_ROUTE)
+async def update_author(data: Author, service: Service, author_id: UUID) -> Author:
+    """Update an author."""
+    return await service.update(author_id, data)
+
+
+@delete(DETAIL_ROUTE, status_code=200)
```
Comment (Goldziher) [inline app/controllers/authors.py:44]: I'd recommend using constants for all status codes

**Review by Goldziher — COMMENTED** [review]
Looks good

File: app/controllers/authors.py (hunk 1 above)
Comment (peterschutt) [reply app/controllers/authors.py:22]:
```suggestion
    """Get a list of authors."""
```

File: app/controllers/authors.py (hunk 2 above)
Comment (peterschutt) [reply app/controllers/authors.py:44]: thx good point

File: tests/integration/test_tests.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+from typing import TYPE_CHECKING
```
Comment (Goldziher) [inline tests/integration/test_tests.py:1]: test_tests? 😉

Comment (peterschutt) [reply tests/integration/test_tests.py:1]: :stuck_out_tongue: just a stub, wanted to see if I could get the pattern working - will replace with something more meaningful soon.

**Pull request merged**